File: EdwinMuraya/BigO/countDown.py
# Time Complexity
# The time complexity is linear
# The number of operation increases proportionaly with increase in the value of input
# The trend would be linear


# Example
# number > 10
# The number of operations are 10 loops
def countDown(n):
    while n > 0:
        print(n)         # O(1)
        n -=1            # O(1)
    print("Blast off")

countDown(5)

# Time Complexity => O(N) + 2 O(1)
# we are interested with the worst case scenario

# Time complexity is therefore O(N) where n is the number passed to the
# function.

File: EdwinMuraya/BigO/CountDown.js
// Time complexity
// The time complexity is Linear since we have a while loop that loops over
// n times. The number of operation increses proportionaly with the increase in the value of the 
// input value n

//For a worst case scenario. 
  // The trend would be increase linearly.

// Examples
countDown(5) // number of operation equal to 5 times
countDown(6) // number of operation equal to 6 times
countDown(1000) // number of operation equal to 1000 times.

function countDown(n){
   while(n > 0){
      console.log(n)
     n-=1;
   }
   console.log("Blast off")
}
